src/pages/home/report/ReportActionItem.js

```diff
@@ -215,108 +198,117 @@ class ReportActionItem extends Component {
                         </View>
-                    )}
-                </Hoverable>
-                <View style={styles.reportActionSystemMessageContainer}>
-                    <InlineSystemMessage message={this.props.action.error} />
-                </View>
-            </PressableWithSecondaryInteraction>
-        );
-    }
+                        <MiniReportActionContextMenu
+                            reportID={props.report.reportID}
+                            reportAction={props.action}
+                            isArchivedRoom={ReportUtils.isArchivedRoom(props.report)}
+                            displayAsGroup={props.displayAsGroup}
+                            isVisible={
+                                hovered
+                                && !props.draftMessage
+                            }
+                            draftMessage={props.draftMessage}
+                            isChronosReport={ReportUtils.chatIncludesChronos(props.report)}
+                        />
+                    </View>
+                )}
+            </Hoverable>
+            <View style={styles.reportActionSystemMessageContainer}>
+                <InlineSystemMessage message={props.action.error} />
+            </View>
+        </PressableWithSecondaryInteraction>
+    );
 }
+
 ReportActionItem.propTypes = propTypes;
 ReportActionItem.defaultProps = defaultProps;
 
-export default compose(
-    withWindowDimensions,
-    withLocalize,
-    withNetwork(),
-    withBlockedFromConcierge({propName: 'blockedFromConcierge'}),
-    withReportActionsDrafts({
-        propName: 'draftMessage',
-        transformValue: (drafts, props) => {
-            const draftKey = `${ONYXKEYS.COLLECTION.REPORT_ACTIONS_DRAFTS}${props.report.reportID}_${props.action.reportActionID}`;
-            return lodashGet(drafts, draftKey, '');
-        },
-    }),
-)(ReportActionItem);
+export default React.memo(
+    compose(
+        withWindowDimensions,
+        withLocalize,
+        withNetwork(),
+        withBlockedFromConcierge({propName: 'blockedFromConcierge'}),
+        withReportActionsDrafts({
+            propName: 'draftMessage',
+            transformValue: (drafts, props) => {
+                const draftKey = `${ONYXKEYS.COLLECTION.REPORT_ACTIONS_DRAFTS}${props.report.reportID}_${props.action.reportActionID}`;
+                return lodashGet(drafts, draftKey, '');
+            },
+        }),
+    )(ReportActionItem),
+    (prevProps, nextProps) => (
+        prevProps.displayAsGroup === nextProps.displayAsGroup
+            && prevProps.draftMessage === nextProps.draftMessage
+            && prevProps.isMostRecentIOUReportAction === nextProps.isMostRecentIOUReportAction
+            && prevProps.hasOutstandingIOU === nextProps.hasOutstandingIOU
+            && prevProps.shouldDisplayNewMarker === nextProps.shouldDisplayNewMarker
+            && _.isEqual(prevProps.action, nextProps.action)
+    ),
+);
```

Comment:
Is this behavior equivalent? Just thinking about where this optimization happened before and it's not quite in the same spot is it?

Reply:
Sorry I'm not really following, can you please elaborate?

Reply:
Before, `shouldComponentUpdate()` applied it's optimizations around `ReportActionItem`. `React.memo()` is applied around `ReportActionItem` after `ReportActionItem` has been wrapped in all the HOCs.